File: rust_dr/src/repl.rs
use std::io::{self, Write};

use crate::lexer::lexer_nom::Lexer;
use crate::lexer::lexer_nom::Token;

pub fn repl() {
    let mut input = String::new();
    println!("Welcome to the Monkey REPL!");
    println!("Feel free to type in commands");
    
    loop {
        print!(">> ");
        io::stdout().flush().unwrap();
        io::stdin().read_line(&mut input).unwrap();
        let mut lex = Lexer::new(input.as_str());
        let mut tok = lex.next_token().unwrap();
        while tok != Token::Eof {
            println!("{}", tok);
            tok = lex.next_token().unwrap();
        }
        input.clear();
    }
}

File: ruby/lib/repl.rb
# frozen_string_literal: true

require_relative "./lexer"

class Repl
  def start
    trap('INT') do
      puts "\nGoodbye!"
      exit
    end

    loop do
      print '>> '
      input = gets.chomp

      l = Lexer.new(input)

      loop do
        tok = l.next_token

        break if tok == Token::Eof

        puts tok.inspect
      end
    end
  end
end
